src/Stage.java

```diff
@@ -25,10 +27,11 @@ public void iterate() {
             // Decode and execute the instruction (You need to implement this part)
             // For simplicity, let's assume the instruction is a no-op (do nothing).
             // You should replace this with actual instruction execution logic.
-
+            decoder.execute(this);
             // Increment the program counter to the next instruction
             pc = nextPc;
-            decoder.execute(this);
+            instructionCount++;
```

Comment:
As intended, instruction Count will not be iterated, but I'll merge your pull-request for now.